src/Testcontainers/Builders/TestcontainersBuilder.cs

```diff
@@ -186,6 +186,9 @@ public ITestcontainersBuilder<TDockerContainer> WithPortBinding(string port, boo
     /// <inheritdoc cref="ITestcontainersBuilder{TDockerContainer}" />
     public ITestcontainersBuilder<TDockerContainer> WithPortBinding(string hostPort, string containerPort)
     {
+      // Prepare Java alignment. Use an empty string instead of 0: https://github.com/docker/for-mac/issues/5588#issuecomment-934600089.
+      hostPort = "0".Equals(hostPort, StringComparison.OrdinalIgnoreCase) ? string.Empty : hostPort;
```

Comment:
Is this the right part in the code to make this change? This would mean, if a user consciously sets it to `0` (not sure if there is a valid reason for this TBH), we would replace it as well.

I would have expected we change all occurrences in our code, where we construct a random mapping to use `""` instead of `0`.

Reply:
> I would have expected we change all occurrences in our code, where we construct a random mapping to use `""` instead of `0`.

Yes, but this would require changes in the modules incl. their extension methods. This change allows us to test the behavior with the least amount of changes.

> But since you mentioned this is a preparation, you maybe wanted to follow up on this after all, then it is of course a good intermediate step as a small change.

Soon as we start to replace the modules, we can remove this line.